pkg/scalers/aws_cloudwatch_scaler.go

```diff
@@ -139,26 +146,35 @@ func parseAwsCloudwatchMetadata(metadata, resolvedEnv map[string]string) (*awsCl
 		return nil, fmt.Errorf("no awsRegion given")
 	}
 
-	accessIDKey := awsAccessKeyIDEnvVar
-	if val, ok := metadata["awsAccessKeyID"]; ok && val != "" {
-		accessIDKey = val
-	}
-
-	if val, ok := resolvedEnv[accessIDKey]; ok && val != "" {
-		meta.awsAccessKeyID = val
+	// Check TriggerAuthentication first
+	if podIdentity == "aws-credentials" {
```

Comment:
This logic is abusing podIdentity - Instead you should determine it yourself based on the parameters that are provided and throw the required errors accordingly.

For example, use `awsRoleArn` if specified. Otherwise use `awsAccessKeyId` & `awsSecretAccessKey`, if those are empty throw an exception